models/tshirt/include/Tshirt.h

```diff
@@ -0,0 +1,289 @@
+
+            return pdm->XHuz - soil_m;
+        }
+
+        //! Calculate the Cschaake, or the Schaake adjusted magic constant by soil type, based on the given Tshirt parameters
+        static double calc_Cschaake(const tshirt_params& params)
+        {
+            return 3.0 * params.satdk / (2.0e-6);
+        }
+
+        /*!
+         * Calculate the height above water table based on the given Tshirt parameters
+         *
+         * Calculate the height above water table based on known constants and the particular `alpha` value given for
+         * relative soil suction head, as provided in the Tshirt parameters.
+         *
+         * @param params
+         * @return
+         */
+        static double calc_Hwt(const tshirt_params& params)
+        {
+            // H_wt = alpha_fc * H_atm ; H_atm = P_atm / gamma ; P_atm = 101,300 [Pa] ; gamma = 9,810 [N m^-3] ('very nearly')
+            return params.alpha_fc * (101300 / 9810);
+        }
+
+        static double calc_Sfc(const tshirt_params& params, const tshirt_state& state)
+        {
+            // TODO: account for possiblity of Hwt being less than 0.5 (though initially, it looks like this will never be the case)
+            double z1 = calc_Hwt(params) - 0.5;
+            double z2 = z1 + 2;
+
+            // Note that z^( 1 - (1/b) ) / (1 - (1/b)) == b * (z^( (b - 1) / b ) / (b - 1)
+            return params.maxsmc * pow((1.0 / params.satpsi), (-1.0 / params.b)) *
+                   ((params.b * pow(z2, ((params.b - 1) / params.b)) / (params.b - 1)) -
+                    (params.b * pow(z1, ((params.b - 1) / params.b)) / (params.b - 1)));
+        }
+
+        static void init_nash_cascade_vector(vector<Nonlinear_Reservoir>& reservoirs, const tshirt_params& params,
+                const tshirt_state& state, const double activation, const double max_flow_velocity)
+        {
+            reservoirs.resize(params.nash_n);
+            for ( unsigned long i = 0; i < reservoirs.size(); ++i )
+            {
+                //construct a single outlet nonlinear reservoir
+                reservoirs[i] = Nonlinear_Reservoir(0, params.Ssmax, state.Snash[i], params.Kn, 1, activation,
+                        max_flow_velocity);
+            }
+        }
+
+        //! run one time step of tshirt
+        static int run(
+                double dt,
+                tshirt_params params,        //!< static parameters for tshirt
+                tshirt_state state,          //!< model state
+                tshirt_state& new_state,     //!< model state struct to hold new model state
+                tshirt_fluxes& fluxes,       //!< model flux object to hold calculated fluxes
+                double input_flux_meters,          //!< the amount water entering the system this time step
+                // TODO: should/can this be a smart pointer?
+                giuh_kernel* giuh_obj,       //!< kernel object for calculating GIUH runoff from subsurface lateral flow
```

Comment:
The catchment realization should hold the giuh kernel (as a smart pointer) and manage the giuh memory.  The realization is what composes these kernels/models together into a realized model.

Reply:
@hellkite500, so only the "raw" surface runoff needs to be present and utilized inside the Tshirt model kernel, with the GIUH calculations happening completely separately from `tshirt_kernel.run(...)`?  

This seems sensible, but the card for creating the Tshirt class suggests having the GIUH kernel directly included (and, for that matter, `tshirt_kernel` _requires_ reservoir kernel class objects).  I.e., in some respects, the separation of concerns between the Tshirt model kernel class versus the Tshirt model realization is unclear.

More practically, it would also hinder a spot I see to employ some concurrency inside the `run` function in the future, though that could be made up for in various ways and may not be a priority.